File: src/main/java/ER/Account.java
package ER;

import java.util.function.Predicate;

public class Account {
    @Check(valid = "nonNegative")
    private long balance;
    @Check(valid = "dotCom")
    private String email;

    public Account(long balance, String email) {
        this.balance = balance;
        this.email = email;
    }

    public long getBalance() {
        return balance;
    }

    public void setBalance(long balance) {
        this.balance = balance;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public boolean nonNegative(long v) {
        return v >= 0;
    }

    public boolean dotCom(String address) {
        return address.endsWith(".com");
    }
}

File: src/main/kotlin/ER/G1/Account.kt
package ER.G1

import ER.Check

class Account {
    var age: Int = 17

    @Check(valid = "nonNegative")
    var balance: Long = 0

    @Check(valid = "dotCom")
    lateinit var email: String

    fun nonNegative(v: Long) = v > 0
    fun dotCom(address: String) = address.endsWith(".com")
}